app/controllers/profiles_controller.rb

```diff
@@ -19,14 +19,7 @@ def update
 
       redirect_to (session.delete(:target) || root_url)
     else
-      if current_user.email == "" && current_user.unconfirmed_email.nil?
-        current_user.errors[:email].clear
-        current_user.errors[:email] = " can't be blank"
-        flash.now[:danger] = "Unable to save profile. Please correct the following: #{current_user.errors.full_messages.join(', ')}."
-      else
-        current_user.errors[:email].clear
-        flash[:danger] = I18n.t("devise.registrations.update_needs_confirmation")
-      end
```

Comment:
Added an additional :email validation for update in the user model, so could remove a lot from this method. Let me know if there's a better way to do this. Thanks.